src/Classes/ItemsTab.lua

```diff
@@ -2266,14 +2309,28 @@ function ItemsTabClass:AddCustomModifierToDisplayItem()
 		item:BuildAndParseRaw()
 		return item
 	end
-	controls.sourceLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 95, 20, 0, 16, "^7Source:")
-	controls.source = new("DropDownControl", {"TOPLEFT",nil,"TOPLEFT"}, 100, 20, 150, 18, sourceList, function(index, value)
-		buildMods(value.sourceId)
+	controls.sourceLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 100, 20, 0, 16, "^7Source:")
+	controls.source = new("DropDownControl", {"TOPLEFT",nil,"TOPLEFT"}, 105, 20, 150, 18, sourceList, function(index, value)
+		currentSourceId = value.sourceId
+		buildMods()
 		controls.modSelect:SetSel(1)
 	end)
 	controls.source.enabled = #sourceList > 1
-	controls.modSelectLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 95, 45, 0, 16, "^7Modifier:")
-	controls.modSelect = new("DropDownControl", {"TOPLEFT",nil,"TOPLEFT"}, 100, 45, 600, 18, modList)
+	controls.modSelectLabel = new("LabelControl", {"TOPRIGHT",nil,"TOPLEFT"}, 100, 45, 0, 16, "^7Modifier:")
+	controls.modSelectLabel.shown = function()
+		local sourceId = sourceList[controls.source.selIndex].sourceId
+		return sourceId == "CUSTOM" or sourceId == "PREFIX" or sourceId == "SUFFIX"
+	end
+	controls.modSelectDropDown = new("DropDownControl", {"TOPRIGHT",nil,"TOPLEFT"}, 99, 45, 90, 18, modifiersDDList , function(index, value)
+		currentModId = value.modId
+		buildMods()
```

Comment:
You really don't need to change the method signature everywhere just to enable this call.  Just grab the sourceId from the other control and add the modId as a second parameter:

```suggestion
		buildMods(controls.source.list[controls.source.selIndex].sourceId, value.modId)
```